app/src/androidTest/java/org/openobservatory/ooniprobe/client/OONIAPIClientTest.java

```diff
@@ -64,6 +72,51 @@ public void onError(String msg) {
     }
 
     @Test
-    public void testDeleteJsons() {
+    public void testSelectMeasurementsWithJson() {
+        Delete.table(Measurement.class);
+        addMeasurement(EXISTING_REPORT_ID, true);
+        addMeasurement(EXISTING_REPORT_ID_2, true);
+        addMeasurement(NONEXISTING_REPORT_ID, true);
+        addMeasurement(NONEXISTING_REPORT_ID, true);
+        List<Measurement> measurements = Measurement.selectMeasurementsWithJson(a);
+        if (measurements.size() == 2 &&
+                containsMeasurement(measurements, EXISTING_REPORT_ID) &&
+                containsMeasurement(measurements, EXISTING_REPORT_ID_2))
+            Assert.assertTrue(true);
+
+        Assert.fail();
```

Comment:
I do not think the above code is correct. I suggest to assert the stuff inside the `if`. Also, it seems weird that after the `if` you fall through straight to failure. Why is that?

Reply:
For this test I need your help, I'm totally stuck